annotations/views/rest_views.py

```diff
@@ -476,25 +479,87 @@ def create(self, request, *args, **kwargs):
         return Response(serializer.data,
                         status=status.HTTP_201_CREATED,
                         headers=headers)
-
     @list_route()
     def search(self, request, **kwargs):
         q = request.GET.get('search', None)
         if not q:
-            return Response({'results': []})
-        pos = request.GET.get('pos', None)
-        concepts = goat.Concept.search(q=q, pos=pos, limit=50)
-
-        def _relabel(datum):
-            _fields = {
-                'name': 'label',
-                'id': 'alt_id',
-                'identifier': 'uri'
-            }
-
-            return {_fields.get(k, k): v for k, v in datum.iteritems()}
-        return Response({'results': map(_relabel, [c.data for c in concepts])})
+                return Response({'results': []})
+        force = request.GET.get('force')
 
+        if cache.get(q.lower()) and q and force == None: # If force is true we want to run the query from scratch and not use the cached results
+            results = cache.get(q.lower())
+        else:
+            pos = request.GET.get('pos', None)
+            concepts = goat.Concept.search(q=q, pos=pos, limit=50)
+
+            def _relabel(datum):
+                _fields = {
+                    'name': 'label',
+                    'id': 'alt_id',
+                    'identifier': 'uri'
+                }
+
+                return {_fields.get(k, k): v for k, v in datum.iteritems() }
+            results = map(_relabel, [c.data for c in concepts])
+
+            def _parseViaf(self, uri, new_concepts):
+                data = urllib2.urlopen(uri)
+                tree = e.parse(data)
+                root = tree.getroot()
+                namespace = {'foaf': 'http://xmlns.com/foaf/0.1/', 'ns1': 'http://viaf.org/viaf/terms#', 'owl': 'http://www.w3.org/2002/07/owl#', 'rdf': 'http://www.w3.org/1999/02/22-rdf-syntax-ns#', 'void': 'http://rdfs.org/ns/void#'}
+                find = e.XPath("//ns1:text", namespaces = namespace)
+                concept_name = find(root)[0].text
+                concept_description = find(root)[0].text
```

Comment:
since you don't use those variables for anything else, just put the xml expression into the dictionary creation